data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -669,6 +756,246 @@ class ApiServerClient {
       })
       .then(resource => _.get(resource, 'spec.options.context'));
   }
+
+  /**
+   * @description Create OSB Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Name of resource group 
+   * @param {string} opts.resourceType - Type of resource 
+   * @param {string} opts.resourceId - Unique id of resource 
+   * @param {string} opts.metadata - Optional; pass finalizers or some other field
+   * @param {string} opts.labels - to be put in label
+   * @param {Object} opts.spec - Value to set for spec field of resource
+   * @param {string} opts.status - status of the resource
+   */
+  createOSBResource(opts) {
+    logger.info(`Creating resource with opts: `, opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to create resource`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to create resource`);
+    assert.ok(opts.resourceId, `Property 'resourceId' is required to create resource`);
+    assert.ok(opts.spec, `Property 'spec' is required to create resource`);
+    const metadata = _.merge(opts.metadata, {
+      name: opts.resourceId
+    });
+    if (opts.labels) {
+      metadata.labels = opts.labels;
+    }
+    const crdJson = this.getCrdJson(opts.resourceGroup, opts.resourceType);
+    const resourceBody = {
+      apiVersion: `${crdJson.spec.group}/${crdJson.spec.version}`,
+      kind: crdJson.spec.names.kind,
+      metadata: metadata,
+      spec: camelcaseKeys(opts.spec)
+    };
+
+    if (opts.status) {
+      const statusJson = {};
+      _.forEach(opts.status, (val, key) => {
+        if (key === 'state') {
+          resourceBody.metadata.labels = _.merge(resourceBody.metadata.labels, {
+            'state': val
+          });
+        }
+        statusJson[key] = _.isObject(val) ? JSON.stringify(val) : val;
```

Comment:
i thought in the new SFServiceInstances/Bindings crds we are not going to have JSON strings within crds, rather they are going to be specific attribs. can we avoid this JSON stringifyication if not required.

Reply:
You are right. We are going with specific object. @amitmalav Can you check this? I guess we can remove it.

Reply:
Thanks. Missed this bit. Have changed now.